inc/compat/plugins/compat-plugin-woo-carrier-agents.php

```diff
@@ -0,0 +1,291 @@
+<?php
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Compatibility with plugin: WooCommerce Carrier Agents (by Markup.fi).
+ */
+class FluidCheckout_WooCarrierAgents extends FluidCheckout {
+
+	/**
+	 * Class name for the plugin which this compatibility class is related to.
+	 */
+	public const CLASS_NAME = 'Woo_Carrier_Agents';
+
+	/**
+	 * Session field name for the selected pickup location.
+	 */
+	public const SESSION_FIELD_NAME = 'carrier-agent';
+
+	/**
+	 * Session field name for the carrier agent data.
+	 */
+	public const SESSION_FIELD_NAME_DATA = 'carrier-agents-data';
+
+
+	/**
+	 *	Carrier agent IDs.
+	 *
+	 * @var array
+	 */
+	public $carrier_agent_ids = array();
+
+
+
+	/**
+	 * __construct function.
+	 */
+	public function __construct() {
+		$this->hooks();
+	}
+
+
+
+	/**
+	 * Initialize hooks.
+	 */
+	public function hooks() {
+		// Postcode search section
+		add_filter( 'woo_carrier_agents_search_output', array( $this, 'maybe_change_postcode_search_placement' ), 10 );
+
+		// Fetch shipping method IDs from the plugin
+		add_action( 'init', array( $this, 'fetch_button_option_values' ), 10 );
+
+		// Persisted data
+		add_action( 'fc_set_parsed_posted_data', array( $this, 'maybe_set_terminals_field_session_values' ), 10 );
+
+		// Maybe set step as incomplete
+		add_filter( 'fc_is_step_complete_shipping', array( $this, 'maybe_set_step_incomplete_shipping' ), 10 );
+
+		// Add substep review text lines
+		add_filter( 'fc_substep_shipping_method_text_lines', array( $this, 'add_substep_text_lines_shipping_method' ), 10 );
+
+		// Checkout validation settings
+		add_filter( 'fc_checkout_validation_script_settings', array( $this, 'change_js_settings_checkout_validation' ), 10 );
+	}
+
+
+
+	/**
+	 * Fetch shipping method IDs from the plugin.
+	 */
+	public function fetch_button_option_values() {
+		// Bail if class is not available
+		if ( ! class_exists( self::CLASS_NAME ) ) { return; }
+
+		// Get object
+		$class_object = FluidCheckout::instance()->get_object_by_class_name_from_hooks( self::CLASS_NAME );
+
+		// Bail if class method is not available
+		if ( ! method_exists( $class_object, 'get_carrier_titles' ) ) { return; }
+
+		$shipping_methods = $class_object->get_carrier_titles();
+
+		// Bail if not array
+		if ( ! is_array( $shipping_methods ) ) { return; }
+
+		// Turn 2 dimensional array into 1 dimensional with keys as values
+		array_walk_recursive( $shipping_methods, function( $value, $key ) {
+			$this->carrier_agent_ids[] = $key;
+		} );
```

Comment:
@vlad-dedikov, wouldn't this be enough to get the array keys?

```php
array_keys( $shipping_methods );
```

Or this, to get the array values instead?
```php
array_values( $shipping_methods );
```

Reply:
Right you are - it was a bit more complex initially but I forgot to simplify it in the end...